### PR TITLE
Raise an exception while creating preset if pretender http mock is expired

### PR DESCRIPTION
**Pull request by IrynaMironava**

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -6,4 +6,4 @@ Core developers:
 Additional contributors:
 * OddBloke - Daniel Watkins
 * ulcooney - Paul Cooney
-* Khol - Igor Khol
+* Khrol - Igor Khrol
```
Comment (txels) [inline AUTHORS]: oh

File: pretenders/server/apps/preset.py [hunk 2]
```diff
@@ -89,6 +91,14 @@ def add_preset(uid):
         raise HTTPResponse(("Preset has {0} times. Must be greater than "
                              "zero.".format(preset.times).encode()),
                            status=400)
+    try:
+        pretender = get_pretenders('http')[uid]
+        if datetime.datetime.now() - pretender.start > pretender.timeout:
```
Comment (txels) [inline pretenders/server/apps/preset.py]: I'd prefer to extract this logic into a function e.g. `pretender.has_expired` in `PretenderModel`.

**Comment by txels** [comment]
Thanks for this. Looks good to me, especially if you relocate that check into the PretenderModel class for better encapsulation.

Maybe @alexcouper has something to add here?

**Comment by alexcouper** [comment]
I believe this covers the case where the maintainer hasn't run yet (it runs every 5 secs) but a pretender has expired. Makes sense to have this.

I agree though regarding abstraction, but I'll make the corresponding change, as we'll want to also change the logic in the delete call that the maintainer uses.

**Comment by alexcouper** [comment]
In future, a test would be nice to go along with the change.

Thanks for taking the time to contribute though :)

**Pull request merged**